modules/phase_field/src/kernels/PFFracBulkRate.C

```diff
@@ -0,0 +1,122 @@
+#include "PFFracBulkRate.h"
+
+template<>
+InputParameters validParams<PFFracBulkRate>()
+{
+  InputParameters params = validParams<KernelValue>();
+  params.addRequiredParam<Real>("l","Interface width");
+  params.addRequiredParam<Real>("visco","Viscosity parameter");
+  params.addRequiredCoupledVar("beta", "Auxiliary variable");
+  params.addCoupledVar("disp_x", "The x displacement");
+  params.addCoupledVar("disp_y", "The y displacement");
+  params.addCoupledVar("disp_z", "The z displacement");
+
+  return params;
+}
+
+PFFracBulkRate::PFFracBulkRate(const std::string & name,
+                               InputParameters parameters):
+  KernelValue(name,parameters),
+  _gc_prop(getMaterialProperty<Real>("gc_prop")),
+  _G0_pos(getMaterialProperty<Real>("G0_pos")),
+  _dG0_pos_dstrain(getMaterialProperty<RankTwoTensor>("dG0_pos_dstrain")),
+  _betaval(coupledValue("beta")),
+  _beta_var(coupled("beta")),
+  _xdisp_coupled(isCoupled("disp_x")),
+  _ydisp_coupled(isCoupled("disp_y")),
+  _zdisp_coupled(isCoupled("disp_z")),
+  _xdisp_var(_xdisp_coupled ? coupled("disp_x") : 0),
+  _ydisp_var(_ydisp_coupled ? coupled("disp_y") : 0),
+  _zdisp_var(_zdisp_coupled ? coupled("disp_z") : 0),
+  _l(getParam<Real>("l")),
+  _visco(getParam<Real>("visco"))
+{
+}
+
+Real
+PFFracBulkRate::computeDFDOP(PFFunctionType type)
+{
+  Real gc = _gc_prop[_qp];
+
+  switch (type)
+  {
+    case Residual:
+    {
+      Real c = _u[_qp];
+      Real x =   _l * _betaval[_qp] + 2.0*(1.0-c) * _G0_pos[_qp]/gc - c/_l;
+
+      return -(( std::abs(x) + x )/2.0)/_visco;
+    }
+    case Jacobian:
+    {
+      Real c = _u[_qp];
+      Real x = _l * _betaval[_qp] + 2.0 * (1.0 - c) * _G0_pos[_qp] / gc - c / _l;
+      Real signx = x > 0.0 ? 1.0 : -1.0;
+      return (signx + 1.0)/2.0 * (2.0 * _G0_pos[_qp]/gc + 1.0/_l)/_visco;
+    }
+  }
```

Comment:
Clang Static analysis will complain that you left off the default case here. Not a big deal, but it's nice to keep things clean.